### PR TITLE
Add an article about decentralized storage solutions

### PR DESCRIPTION
**Pull request by bucanero**
Fixes https://github.com/near/docs/issues/683

### CODE REVIEW

File: docs/concepts/storage.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+To learn more about Arweave, check its [mining mechanism](https://arwiki.wiki/#/en/arweave-mining) and its [bandwidth-sharing system](https://arwiki.wiki/#/en/karma). 
+:::
+
+## IPFS
+
+The [InterPlanetary File System](https://ipfs.io/) (IPFS) is a protocol and peer-to-peer network for storing and sharing data in a distributed file system. IPFS uses content-addressing to uniquely identify each file in a global namespace connecting all computing devices.
+
+### Content identifier
+
+When you add a file to IPFS, your file is split into smaller chunks, cryptographically hashed, and given a unique fingerprint called a content identifier (CID).
```
Comment (BenKurrek) [inline docs/concepts/storage.md:34]:
```suggestion
When you add a file to IPFS, it is split into smaller chunks, cryptographically hashed, and given a unique fingerprint called a content identifier (CID). 
```

File: docs/concepts/storage.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+When you add a new version of your file to IPFS, since the cryptographic hash is different, it will get a new CID.
+This means that any changes to a file don't overwrite the original, and common chunks across files can be reused in order to minimize storage costs.
+
+### Naming system
+
+IPFS offers a decentralized naming system so you don't need to remember a long string of CIDs.
+IPFS can find the latest version of your file using the IPNS decentralized naming system, and you can use DNSLink to map CIDs to human-readable DNS names.
+
+### IPFS providers
```
Comment (BenKurrek) [inline docs/concepts/storage.md:57]: Is it possible to make all these headers one size down? It's kinda hard to notice the difference between the main sections (SIA, IPFS, ARWEAVE), and their subheadings as shown here: 
image

File: docs/concepts/storage.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+- [Web3.Storage](https://web3.storage/): it's a free service that simplifies building on top of IPFS and Filecoin. Web3.Storage is backed by Filecoin and makes content available via IPFS leveraging the unique properties of each network.
+- [NFT.Storage](https://nft.storage/): this free service is built specifically for storing off-chain NFT data. Data is stored decentralized on IPFS and Filecoin, and referenced using content-addressed IPFS URIs that can be used in your smart contracts.
+
+## Sia
+
+[Sia](https://sia.tech/) is a decentralized cloud storage platform that leverages blockchain technology to create a robust data storage marketplace that is more affordable than traditional cloud storage providers.
+
+### Privacy
+
+Sia encrypts and distributes your files across a decentralized network. With Sia you control the private encryption keys and you own the data, so no third-party can access or control your files.
```
Comment (BenKurrek) [inline docs/concepts/storage.md:68]:
```suggestion
Sia encrypts and distributes your files across a decentralized network. With Sia, you control the private encryption keys and you own the data, so no third-party can access or control your files.
```

File: website/sidebars.json [hunk 4]
```diff
@@ -9,7 +9,8 @@
       "concepts/gas",
       "concepts/indexer",
       "concepts/networks",
-      "concepts/transaction"
+      "concepts/transaction",
+      "concepts/storage-solutions"
```
Comment (thisisjoshford) [inline website/sidebars.json:13]: This looks a little confusing in the sidebar as we have multiple sections that talk about storage. As a new user, I think I would be a little lost when trying to navigate this. Let's have a sub-menu for "Data Storage" that has two parts. "Storage on NEAR" and "Alternative Solutions" or something like this.

**Review by thisisjoshford — CHANGES_REQUESTED** [review]

**Review by thisisjoshford — CHANGES_REQUESTED** [review]
Great overview of these three solutions. I think we should add simple examples of their implementation. Perhaps @doriancrutcher could assist with this?

File: website/sidebars.json (hunk 4 above)
Comment (bucanero) [reply website/sidebars.json]: I've grouped the articles together, under a "Data Storage" category:
- Data Storage
   - Storage on NEAR
   - Storage Staking
   - Alternative Solutions

**Comment by bucanero** [comment]
> Great overview of these three solutions. I think we should add simple examples of their implementation. Perhaps @doriancrutcher could assist with this?

hi @thisisjoshford, I have updated the draft and added examples for each storage (thanks to @doriancrutcher !!)

let me know what you think 😃

**Pull request merged**